[13.0][MIG] account_banking_sepa_credit_transfer: Migration to 13.0

ValentinVinagre opened the pull request:
This PR supersedes https://github.com/OCA/bank-payment/pull/671

- [x] Depends https://github.com/OCA/bank-payment/pull/700

alexis-via committed:
account_banking_sepa_credit_transfer implementing SEPA SCT pain

Commit:
[IMP] account_banking_sepa_credit_transfer: 3 things

* Small API changes
* Updated two additional imports to new API
* A payment mode type needs a suitable bank type

Commit:
Update strings to prepare the arrival of SEPA Direct Debit XSD validation of the SEPA XML file is now really working Update views to 7.0-style Better error handling when Bank account in missing on the invoice.

sbidoul committed:
[RFR] account_banking_payment: refactoring to make payment export features available without pulling bank statement features

- introduce account_banking_payment_export module with all features from account_banking_payment that do not involve changing the payment order workflow or depend on bank statement changes
- account_banking_payment_export does not depend on account_banking
- account_banking_payment is consequently stripped of above features
- account_banking_sepa_credit_transfer depend on account_banking_payment_export and is compatible with the default payment order workflow
- clarified comment about workflow in account_banking_sepa_credit_transfer
- avoid generating invalid sepa credit transfer files when bank accounts have no BIC

Commit:
Added translations to all banking modules in use in the Netherlands

Commit:
Retrieve current date in user's timezone [FIX] Default to current date instead of tomorrow

Commit:
Translation template files for all modules.

Commit:
[IMP] account_banking_sepa_credit_transfer: Support new format + other things:

* Add support for pain.001.001.05 Fix dependencies (I need the field "payment_order_type" which is in account_banking_payment) Only use ASCII characters in the XML file (that's how banks use XML !) Use the payment order reference as the Message Identification of the XML file. Use the payment line sequence as the "End to End Identification" Remove the Instruction Identification ('InstrId') because it is not a required field (only the End to End identification is required) Rename variables in the function create_sepa() to include the number of each field as used in the EPC guidelines Reduce the number of flake8 warnings Make code more modular : - self.pool.get('banking.export.sepa').create() has a _prepare fonction - most of the content of the XML file can be inherited via _prepare_field()
* Update strings. Update FR translation.
* fields.date.context_today doesn't work with datetime => we now use the native field create_date Coding style improvements.
* Add logo

Commit:
Launchpad automatic translations update.

Commit:
[IMP] account_banking_sepa_credit_transfer:

* Better prototype for the _prepare_field function. Update help message, POT file and FR translation.
* Revert the dependancy to account_banking_payment_export (the field payment_order_type on payment.mode.type has a default value of "payment" anyway).
* Akretion-Noviat code sprint dated 2013-11-21: - add module account_banking_pain_base - add support for initiating party identification, priority and structured remittance info in XML file - the requested execution date now uses the fields date_prefered and date_scheduled (the field of the wizard has been removed) - the 'convert to ascii' feature is now an option of the payment mode (enabled by default) - set 'communication' field of payment.line to 140 chars and hide field 'communicatio n2'
* FIX label : Customer (which is wrong, it is most of the time a Supplier !)-> Creditor More PEP8 stuff.

Commit:
Launchpad automatic translations update.

Commit:
[RFR] account_banking_sepa_credit_transfer

* Start code factoring between SCT and SDD.
* More code factoring between SCT and SDD As a consequence, we now have support for structured remittance info in SDD.
* The requested collection date now uses the fields date_prefered and date_scheduled of payment.order, instead of the field in the SDD wizard.
* Mutualize more code between SCT and SDD.
* Replace unallowed ascii caracters by '-' Update some error messages Update translation files and FR translation Include sepa_export in gen_args Factorize more code between SDD and SCT Fix view of payment lines The modules account_banking_pain_base and account_banking_sepa_* are now fully PEP8 compliant
* For SCT, if BIC is not provided, we should not put the 'Creditor Agent' block at all, according to EPC guidelines (on this point, it is different from the Direct Debit !)
* Add demo data
* Add missing utf-8 header Update state field on SEPA file objects and display it in tree+form views. Migrate form views to version 7.0 and simplify them.

Commit:
Launchpad automatic translations update.

pedrobaeza committed:
[MIG] account_banking_sepa_credit_transfer: Migration to v8

sbidoul committed:
sepa credit tfr: get BIC from bank account if not available on bank

Since the bank is optional on bank accounts and the BIC can be set
on the bank account if there is no bank, we allow the system to
get the BIC from the bank account in case it is not on the bank.

We give priority to the BIC indicated on the bank, keeping
the same behaviour with existing data where the bank BIC is
available.

Commit:
Fix an error when no payment_order_ids

Commit:
Use attachments instead of creating a specific object for each kind of payment order

phschmidt committed:
Add requested descriptions in an OCA README.rst file

alexis-via committed:
Port SEPA modules to new API

Fix an important regression in account_banking_sepa_direct_debit: "Date of Last Debit" was not set any more
Proper write of date_done with account_banking_payment_export is installed without account_banking_payment_transfer
Add post-install script for date_sent on payment.order

sbidoul committed:
sepa attachments migration fix

Sometimes sepa file could have been detached
from the corresponding payment order.

Commit:
German pain.001.003.03 format

alexis-via committed:
Add bank.payment.lines object to allow grouping in the payments

pedrobaeza committed:
[IMP] account_banking_sepa_credit_transfer: Short headers + Bump version numbers + es translations

antespi committed:
Define SEPA identifiers per payment mode

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
[MIG] account_banking_sepa_credit_transfer: Migration to 9.0

pedrobaeza committed:
OCA Transbot updated translations from Transifex

Commit:
[MIG] account_banking_sepa_credit_transfer: Migration to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Add support for ISO20022 "Category Purpose"

Add local instrument 'INST' for SEPA Instant Credit Transfer

etobella committed:
[MIG] account_banking_sepa_credit_transfer: Migration to 11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] account_banking_sepa_credit_transfer: Fix tests due to upstream change

Odoo has added a constraint for avoiding a company currency change if there
are move lines, making these tests to fail, as the currency is changed to EUR.

With this commit, we create a new company with EUR currency for avoiding the
problem. An alternative solution could be to delete all the existing move lines
on the company, but as that can be problematic if there are additional
constraints (extra modules, locking date, etc), I have chosen the other approach.

sbidoul committed:
[ADD] sepa: support for Purpose Code

andreparames committed:
[ADD] sepa: add Instruction Identification field

InstrId is optional, so there should be no harm in
adding it always, with the same value as EndToEndId.

chrispi-ch committed:
Translated using Weblate (German)

Currently translated at 100.0% (6 of 6 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_banking_sepa_credit_transfer
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_banking_sepa_credit_transfer/de/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[MIG] account_banking_pain_base: Migration to 12.0

* Standard migration procedure
* README by fragments
* Fixed tests
* Change
* Fix related no readonly in account_banking_pain_base
* Override default partner bank account check on account_payment_partner
* Convert Python date to string

oca-travis committed:
[UPD] Update account_banking_sepa_credit_transfer.pot

oca-travis committed:
[UPD] Update account_banking_sepa_credit_transfer.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

ValentinVinagre committed:
[IMP] account_banking_sepa_credit_transfer: black, isort, prettier

ValentinVinagre committed:
[MIG] account_banking_sepa_credit_transfer: Migration to 13.0

sergio-teruel commented:
@ValentinVinagre Please can you check travis??

ValentinVinagre commented:
Hi @sergio-teruel I'm waiting this fix https://github.com/OCA/bank-payment/pull/700 to be approved.

sergio-teruel commented:
Ok. Thanks.. @carlosdauden Can you review https://github.com/OCA/bank-payment/pull/700?

ValentinVinagre commented:
Hi @sergio-teruel @HaraldPanten full green :)

HaraldPanten commented:
I think there's a kind of "bug" with runbots these days. I've seen it in other PR's. Runbot is working but it fails. And when you try to check all logs a 404 error appears. 

Look at https://github.com/OCA/sale-workflow/pull/1154

@pedrobaeza do you know if we are missing something? Any clue of how could we solve this? Thanks in advance.

ValentinVinagre commented:
Hi @sergio-teruel, @pedrobaeza 
Do you know any solution for the runbot not to get error 404?
Thanks,

pedrobaeza commented:
I can't say. Other times, rebuilding several times has made it.

kurogeek commented:
I just checked on the runbot. I tried to install the "account_banking_sepa_credit_transfer" on runbot system. There was an error about missing module "account_banking_pain_base". I think you might need to include the module in your repo.

![DeepinScreenshot_select-area_20200609100947](https://user-images.githubusercontent.com/7752934/84102099-bda84200-aa39-11ea-913f-fbcb106ca7ae.png)

ValentinVinagre commented:
> I just checked on the runbot. I tried to install the "account_banking_sepa_credit_transfer" on runbot system. There was an error about missing module "account_banking_pain_base". I think you might need to include the module in your repo.
> 
> ![DeepinScreenshot_select-area_20200609100947](https://user-images.githubusercontent.com/7752934/84102099-bda84200-aa39-11ea-913f-fbcb106ca7ae.png)

Done, thanks :)

ValentinVinagre commented:
@HaraldPanten could you review please?

HaraldPanten reviewed (commented):
It's functionally working fine. But I'd like to comment:

As invoices are now account.moves, user must set a "reference" field in invoices, or "communication" field in bank.payment.lines wouldn't be set. If communication fields are not set, an error is occurring as a mandatory field hasn't been set. 

In previous versions, communication field was inherited from invoice numbers if I'm right. But now, it inherits the "reference" field.

Do you think that any info should be shown in invoices when user selects a payment mode related to a "sepa_credit_transfer" payment method? I was thinking about that, as I don't think we should set "reference" field as mandatory.

See attached images:

<img width="1331" alt="Captura de pantalla 2020-06-09 a las 12 55 19" src="https://user-images.githubusercontent.com/32449357/84139759-bf313480-aa50-11ea-9886-69aa6edf3c11.png">

This is my suggestion:

<img width="1422" alt="Captura de pantalla 2020-06-09 a las 12 44 47" src="https://user-images.githubusercontent.com/32449357/84139839-e7b92e80-aa50-11ea-87cb-b4e96a8f98e8.png">


What do you think @ValentinVinagre @pedrobaeza @sergio-teruel ?

THX

ValentinVinagre commented:
Hi @pedrobaeza , @HaraldPanten ,@sergio-teruel 
I think that the problem is here, for some reason the code detects the 'ref' field isn't empty, and doesn't choose the "name" field.

https://github.com/OCA/bank-payment/blob/080bad77d6eedb031004a993743ebf5356821102/account_payment_order/models/account_move_line.py#L52-L64

carlosdauden commented on account_banking_sepa_credit_transfer/tests/test_sct.py:
```diff
@@ -0,0 +1,366 @@
+            }
+        )
+        cls.env.ref(
+            "l10n_generic_coa.configurable_chart_template"
```
Can we use this without having l10n_generic_coa dependency in the manifest?

ValentinVinagre replied:
I think it is correct as it is, if it is added in the manifest the "l10n_generic_coa" will always be installed but it is only used in the test.

pedrobaeza replied:
No, it's not, as this is pre-configuring for the first company that CoA there, provoking problems. Why previous instruction is not working?

ValentinVinagre replied:
I don't understand, should I change this?

pedrobaeza replied:
What is the problem with the code in v12?

ValentinVinagre replied:
IMO the code is correct, this code fragment is the same in v.12. If it isn't correct, please explain more the error, because I don't understand the previous comments.

pedrobaeza replied:
You're right. It was like that in v12, but it's not the best approach, as the module may be missing and this will fail. Let's keep it this way until further analysis.

HaraldPanten commented:
> It's functionally working fine. But I'd like to comment:
> 
> As invoices are now account.moves, user must set a "reference" field in invoices, or "communication" field in bank.payment.lines wouldn't be set. If communication fields are not set, an error is occurring as a mandatory field hasn't been set.
> 
> In previous versions, communication field was inherited from invoice numbers if I'm right. But now, it inherits the "reference" field.
> 
> Do you think that any info should be shown in invoices when user selects a payment mode related to a "sepa_credit_transfer" payment method? I was thinking about that, as I don't think we should set "reference" field as mandatory.
> 
> See attached images:
> 
> <img alt="Captura de pantalla 2020-06-09 a las 12 55 19" width="1331" src="https://user-images.githubusercontent.com/32449357/84139759-bf313480-aa50-11ea-9886-69aa6edf3c11.png">
> 
> This is my suggestion:
> 
> <img alt="Captura de pantalla 2020-06-09 a las 12 44 47" width="1422" src="https://user-images.githubusercontent.com/32449357/84139839-e7b92e80-aa50-11ea-87cb-b4e96a8f98e8.png">
> 
> What do you think @ValentinVinagre @pedrobaeza @sergio-teruel ?
> 
> THX



> Hi @pedrobaeza , @HaraldPanten ,@sergio-teruel
> I think that the problem is here, for some reason the code detects the 'ref' field isn't empty, and doesn't choose the "name" field.
> 
> https://github.com/OCA/bank-payment/blob/080bad77d6eedb031004a993743ebf5356821102/account_payment_order/models/account_move_line.py#L52-L64

Hi @pedrobaeza @carlosdauden what do you think about these suggestions?

Thanks!

HaraldPanten commented:
Could you have a look at our last comments @pedrobaeza @carlosdauden ?

THX!

pedrobaeza committed:
[FIX] account_banking_sepa_direct_debit: Remove l10n_generic_coa need on tests

Tests assume that `l10n_generic_coa` is installed, which is not the case for several
cases. We replace it creating a minimal account base data for the company test that
makes it to work.

pedrobaeza commented:
@ValentinVinagre please give me permissions to this branch for pushing the solution for `l10n_generic_coa` hidden dependency.

pedrobaeza commented:
About the communication, it shouldn't be handled here, as it's something independent from this module, but a general question on `account_payment_order`. Please put an issue with the detailed steps to reproduce the problem.

ValentinVinagre commented:
> ase give me permissions to this branch for pushing the solution for `l10n_generic_coa` hidden dependency.

Done.

pedrobaeza commented:
@sergio-teruel @carlosdauden now it's independent from `l10n_generic_coa`.

HaraldPanten commented:
> About the communication, it shouldn't be handled here, as it's something independent from this module, but a general question on `account_payment_order`. Please put an issue with the detailed steps to reproduce the problem.

Done 

Related issue, RFO: https://github.com/OCA/bank-payment/issues/719

pedrobaeza commented:
/ocabot merge nobump
